db: Handle NULL storage_volume description column in patch 34

freeekanayaka opened the pull request:
Might help with #7857, although I'm not sure since we don't have the database tarball yet.

In any case, this change seems to make sense regardless, since strictly speaking the `SELECT *` in the patch is not guaranteed to return columns in a specific order, and the `description` column doesn't have a `NOT NULL` constraint, so we need to coalesce it.

Signed-off-by: Free Ekanayaka <free.ekanayaka@canonical.com>

tomponline reviewed (approved):
Makes sense, yes `select *` is best avoided in my experience.

freeekanayaka commented:
I've added a unit test too, for good measure. Doesn't exercise the full patch logic, which is complex, but at least it does exercise the bit I changed.

lxc-jenkins commented:
Testsuite passed
